internal/rbd/replicationcontrollerserver.go

```diff
@@ -546,6 +551,32 @@ func (rs *ReplicationServer) PromoteVolume(ctx context.Context,
 	return &replication.PromoteVolumeResponse{}, nil
 }
 
+// checkHealthyPrimary checks if the image is in up+stopped state.
+func checkHealthyPrimary(ctx context.Context, rbdVol *rbdVolume) error {
+	mirrorStatus, err := rbdVol.getImageMirroringStatus()
+	if err != nil {
+		return err
+	}
+	localStatus, err := mirrorStatus.LocalStatus()
+	if err != nil {
+		// LocalStatus can fail if the local site status is not found in
+		// mirroring status. Log complete sites status to debug why getting
+		// local status failed
+		log.ErrorLog(ctx, "mirroring status is %+v", mirrorStatus)
+
+		return fmt.Errorf("failed to get local status: %w", err)
+	}
+
+	if !localStatus.Up && localStatus.State != librbd.MirrorImageStatusStateStopped {
+		return fmt.Errorf("%s is not healthy. Image state is up=%t, state=%q",
```

Comment:
There is no `%w` here, can there be an error that callers can check with `errors.Is()`?

Reply:
AFAIK I don't think it's needed. no caller is going to extract the error message for comparison. this is directly sent to GRPC

Reply:
It was just a suggestion, not a requested change. By using `%w` and pre-defined errors functions are easier to re-use. It might be useful in the future, hence the consideration.

Reply:
added new error and returned with `%w` for future.